src/org/traccar/reports/Events.java

```diff
@@ -19,12 +19,13 @@ private Events() {
 
     public static String getJson(long userId, Collection<Long> deviceIds, Collection<Long> groupIds,
             Collection<String> types, Date from, Date to) throws SQLException {
-        JsonObjectBuilder json = Json.createObjectBuilder();
+        JsonArrayBuilder json = Json.createArrayBuilder();
         for (long deviceId: ReportUtils.getDeviceList(deviceIds, groupIds)) {
             Context.getPermissionsManager().checkDevice(userId, deviceId);
             for (String type : types) {
-                json.add(String.valueOf(deviceId), JsonConverter.arrayToJson(Context.getDataManager()
```

Comment:
It worked but did over-complicating in web part. Invent new model with `deviceId` and array and other things. Bare array of objects is easier and still can be filtered by `deviceId` on `Ext.store` side.

`ArrayBuilder` with `arrayToJson` for every device also did not fit because it produced array of arrays